src/lib/controllers/aligned-controller.js

```diff
@@ -11,23 +15,54 @@ export default class AlignedController {
       console.error(L10nSingleton.getMsgS('ALIGNED_CONTROLLER_NOT_READY_FOR_TOKENIZATION'))
       return false
     }
+
     this.alignment = alignment
     const tokenizer = 'simpleWordTokenization'
-    return this.alignment.createAlignedTexts(tokenizer)
+    const result = this.alignment.createAlignedTexts(tokenizer)
+
+    const res2 = this.alignment.equalSegmentsAmount
+
+    if (!res2) {
+      console.error(L10nSingleton.getMsgS('ALIGNED_CONTROLLER_NOT_EQUAL_SEGMENTS'))
+      this.alignment.clearAlignedTexts()

```

Comment:
We could have here two workflows on segments equality fail:
  - clear everything
  - a user could change any source text he/she likes , we don't track changes
   - align again from scratch on Align button click

or 
   - clear only wrong targets texts (because we should choose some of them that we consider to be correct, by default it could be an origin)
   -  freeze correct origin and targets and allow a user to change only the text that differs by segments amount
   - align again (and may be return again to the start)
or
   - clear only wrong targets texts
   - don't freeze and track changes for all texts 
   - if any changes would be done - clear aligned texts
   - align again (and may be return again to the start)

From my point of view the first one is the most easier and consistent.
If we want to reduce tokenization re-done, than it would be better to add simple segments checks to the tokenization service.
@balmas , if you think that some of the worklows is better - please tell me and I will change it (may be it is better to create a specific issue for that and do it through a new PR)? what do you think?

Reply:
Well, I think the workflow requirements could be different depending upon whether this is upon initial creation of the alignment, or upon adding a translation text after alignment has started.  I think currently the code doesn't support the latter, but I have noted elsewhere in the review that it should and could be done as a separate item. So yes, we could put this requirement on that item :-)